# Assignment 1: Data and Question(s)

## Group 24: Julian Widjaja

### Data:

We decided to use this dataset for our project: https://archive.ics.uci.edu/dataset/275/bike+sharing+dataset

This dataset contains the hourly and daily count of rental bikes between years 2011 and 2012 in Capital bikeshare system with the corresponding weather and seasonal information.

Fanaee-T (2013) collected the data from Capital bikeshare system in Washington D.C. during the 2011-2012 year period.

In [1]:
day_link <- "https://raw.githubusercontent.com/Julian-UBC/stat-301-project/main/data/day.csv"
hour_link <- "https://raw.githubusercontent.com/Julian-UBC/stat-301-project/main/data/hour.csv"

data_day <- read.csv(day_link, header = TRUE, sep = ",")
data_hour <- read.csv("data/hour.csv", header = TRUE, sep = ",")

`data_day` contains the daily count data. Below we can see the first 6 rows of `data_day`.

In [2]:
head(data_day)

instant dteday     season yr mnth holiday weekday workingday weathersit
1 1       2011-01-01 1      0  1    0       6       0          2         
2 2       2011-01-02 1      0  1    0       0       0          2         
3 3       2011-01-03 1      0  1    0       1       1          1         
4 4       2011-01-04 1      0  1    0       2       1          1         
5 5       2011-01-05 1      0  1    0       3       1          1         
6 6       2011-01-06 1      0  1    0       4       1          1         
  temp     atemp    hum      windspeed casual registered cnt 
1 0.344167 0.363625 0.805833 0.1604460 331     654        985
2 0.363478 0.353739 0.696087 0.2485390 131     670        801
3 0.196364 0.189405 0.437273 0.2483090 120    1229       1349
4 0.200000 0.212122 0.590435 0.1602960 108    1454       1562
5 0.226957 0.229270 0.436957 0.1869000  82    1518       1600
6 0.204348 0.233209 0.518261 0.0895652  88    1518       1606

`data_hour` contains the hourly count data. Below we can see the first 6 rows of`data_hour`.

In [3]:
head(data_hour)

instant dteday     season yr mnth hr holiday weekday workingday weathersit
1 1       2011-01-01 1      0  1    0  0       6       0          1         
2 2       2011-01-01 1      0  1    1  0       6       0          1         
3 3       2011-01-01 1      0  1    2  0       6       0          1         
4 4       2011-01-01 1      0  1    3  0       6       0          1         
5 5       2011-01-01 1      0  1    4  0       6       0          1         
6 6       2011-01-01 1      0  1    5  0       6       0          2         
  temp atemp  hum  windspeed casual registered cnt
1 0.24 0.2879 0.81 0.0000    3      13         16 
2 0.22 0.2727 0.80 0.0000    8      32         40 
3 0.22 0.2727 0.80 0.0000    5      27         32 
4 0.24 0.2879 0.75 0.0000    3      10         13 
5 0.24 0.2879 0.75 0.0000    0       1          1 
6 0.24 0.2576 0.75 0.0896    0       1          1

In [4]:
nrow(data_hour)
nrow(data_day)

[1] 17379

[1] 731

`data_hour` has a total of 17379 observations with no missing values. `data_day` has a total of 731 observations with no missing values.

`data_hour` has 17 variables while `data_day` has 16 variables. Both `data_hour` and `data_day` have the same following variables, except `hr` which is only in `data_hour`.

- instant (int): record index
- dteday (chr): date
- season (int): season (1:winter, 2:spring, 3:summer, 4:fall)
- yr (int): year (0: 2011, 1:2012)
- mnth (int): month (1 to 12)
- hr (int): hour (0 to 23) [only for `data_hour`]
- holiday (int): whether day is holiday or not (extracted from http://dchr.dc.gov/page/holiday-schedule)
- weekday (int): day of the week
- workingday (int): if day is neither weekend nor holiday is 1, otherwise is 0.
- weathersit (int): 
    - 1: Clear, Few clouds, Partly cloudy, Partly cloudy
    - 2: Mist + Cloudy, Mist + Broken clouds, Mist + Few clouds, Mist
    - 3: Light Snow, Light Rain + Thunderstorm + Scattered clouds, Light Rain + Scattered clouds
    - 4: Heavy Rain + Ice Pallets + Thunderstorm + Mist, Snow + Fog
- temp (dbl): Normalized temperature in Celsius. The values are derived via (t-t_min)/(t_max-t_min), t_min=-8, t_max=+39 (only in hourly scale)
- atemp (dbl): Normalized feeling temperature in Celsius. The values are derived via (t-t_min)/(t_max-t_min), t_min=-16, t_max=+50 (only in hourly scale)
- hum (dbl): Normalized humidity. The values are divided to 100 (max)
- windspeed (dbl): Normalized wind speed. The values are divided to 67 (max)
- casual (int): count of casual users
- registered (int): count of registered users
- cnt (int): count of total rental bikes including both casual and registered



**Question:**
Clearly state the question you will try to answer using the selected dataset. Your question should involve one random variable of interest (the response) and one or more explanatory variables. Describe clearly how the data will help you address the question of interest. Explain whether your question is focused on prediction, inference, or both.

It is fine to have the same question as other group members. However, you don’t need to agree on a unique common question for the group project. In fact, usually many questions can be answered with the same dataset. Regardless of how many questions are proposed within each group, each team member needs to state and justify a question of interest.

### Question:

What is the expected total count of rental bikes daily based on season, working day, and temperature?

The data could help us answer this question because they have the variables needed to answer our question in each observation. In this case, our response variable is `cnt` and our explanatory variables are `season`, `temp`, and `workingday`. 

### References

Fanaee-T,Hadi. (2013). Bike Sharing Dataset. UCI Machine Learning Repository. https://doi.org/10.24432/C5W894.